client/allocrunner/taskrunner/envoy_bootstrap_hook.go

```diff
@@ -291,22 +328,24 @@ func (h *envoyBootstrapHook) Prestart(ctx context.Context, req *ifs.TaskPrestart
 		cmd.Env = bootstrapEnv
 
 		// Redirect stdout to secrets/envoy_bootstrap.json.
-		fd, fileErr := os.Create(bootstrapFilePath)
+		stdout, fileErr := os.Create(bootstrapFilePath)
 		if fileErr != nil {
 			return false, fmt.Errorf("failed to create secrets/envoy_bootstrap.json for envoy: %w", fileErr)
 		}
-		cmd.Stdout = fd
+		defer stdout.Close()
+		cmd.Stdout = stdout
 
-		// Redirect stderr into a buffer for later reading.
-		buf := bytes.NewBuffer(nil)
-		cmd.Stderr = buf
+		// Redirect stderr into another file for later debugging.
+		stderr, fileErr := os.OpenFile(bootstrapStderrPath, os.O_RDWR|os.O_CREATE, 0666)
```

Comment:
does this need to be `0666` and not `0644`?

Reply:
Sure, changed it.